### PR TITLE
pkg/tinydtls: add IPv4 support

### PR DESCRIPTION
**Pull request by HendrikVE**
### Contribution description

Add IPv4 support to `pkg/tinydtls`.

This PR contains two patches which I will try to get merged in the upstream repository once this PR is merged.

### Testing procedure

Can be tested with #17763

### Issues/PRs references

Dependencies:
- #17762

### CODE REVIEW

File: pkg/tinydtls/patches/0002-Rename-condition-WITH_RIOT_GNRC-to-WITH_RIOT.patch [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+diff --git a/session.c b/session.c
+index 300224e..8e9f544 100644
+--- a/session.c
++++ b/session.c
+@@ -31,7 +31,7 @@
+    && (A)->port == (B)->port					\
+    && uip_ipaddr_cmp(&((A)->addr),&((B)->addr))			\
+    && (A)->ifindex == (B)->ifindex)
+-#elif defined(WITH_RIOT_GNRC)
++#elif defined(WITH_RIOT)
```
Comment (benpicco) [inline pkg/tinydtls/patches/0002-Rename-condition-WITH_RIOT_GNRC-to-WITH_RIOT.patch:23]:
```suggestion
#elif defined(RIOT_VERSION)
````

should work as well and doesn't require a custom define (this is used by other pkgs too)

Comment (benpicco) [reply pkg/tinydtls/patches/0002-Rename-condition-WITH_RIOT_GNRC-to-WITH_RIOT.patch:23]: Ah wait so when RIOT is used with the `posix_sockets` module this path should *not*  be taken - then how about `WITH_RIOT_SOCK`

**Comment by benpicco** [comment]
CI is unhappy

File: pkg/tinydtls/patches/0002-Rename-condition-WITH_RIOT_GNRC-to-WITH_RIOT.patch (hunk 1 above)
Comment (HendrikVE) [reply pkg/tinydtls/patches/0002-Rename-condition-WITH_RIOT_GNRC-to-WITH_RIOT.patch]: I renamed `WITH_RIOT_GNRC` to `WITH_RIOT_SOCK` as suggested.

**Comment by HendrikVE** [comment]
I fixed the compilation for `examples/dtls-echo`. It does not make use of IPv4 yet. The failed test is unrelated to these changes, otherwise CI seems happy now.

**Review by benpicco — APPROVED** [review]
Looks good to me.
Btw: tinydtls has RIOT support, so you also want to submit your patch [upsream](https://github.com/eclipse/tinydtls) so we can drop it in the future.

**Comment by HendrikVE** [comment]
I created PRs accordingly:
- https://github.com/eclipse/tinydtls/pull/128 
- https://github.com/eclipse/tinydtls/pull/129

**Comment by HendrikVE** [comment]
Thank you very much for your review @benpicco !

**Comment by miri64** [comment]
This PR broke `examples/dtls-sock`. See https://github.com/RIOT-OS/RIOT/pull/17765.

**Pull request merged**